lib/modules/manager/composer/utils.ts

```diff
@@ -75,7 +75,18 @@ export function extractConstraints(
 
   // extract php
   if (composerJson.config?.platform?.php) {
-    res.php = composerJson.config.platform.php;
+    const major = api.getMajor(composerJson.config.platform.php);
+    const minor = api.getMinor(composerJson.config.platform.php);
+    const patch = api.getPatch(composerJson.config.platform.php);
+    if (patch === null || patch < 1) {
```

Comment:
patch will never be negative.

Reply:
What's returned is a number, so this handles a float of 0.0 also. But if the return is always an integer, I can update this line to be:
```ts
if (patch === null || patch === 0) {
```